include/unicorn/unicorn.h

```diff
@@ -612,6 +640,7 @@ See sample_ctl.c for a detailed example.
 #define uc_ctl_request_cache(uc, address, tb)                                  \
     uc_ctl(uc, UC_CTL_READ_WRITE(UC_CTL_TB_REQUEST_CACHE, 2), (address), (tb))
 #define uc_ctl_flush_tlb(uc) uc_ctl(uc, UC_CTL_WRITE(UC_CTL_TB_FLUSH, 0))
+#define uc_ctl_tlb_mode(uc, mode) uc_ctl(uc, UC_CTL_WRITE(UC_CTL_TLB_TYPE, 1), (mode))
```

Comment:
Could you illustrate more on this in this file?

Reply:
Ok, in general I only add the necessary types and defines. To avoid to restructuring the complete file I moved some typedefs forword.

But step by step. First the two typedefs (uc_tlb_entry and uc_mem_type). They are necessary before the callback typedef, because the callback has these types as parameter. But I wanted to keep the structure of the file so I moved only the typedef and not the complete struct/enum definition.

Next the `uc_cb_tlbevent_t` typedef, a functionpointer type for the tlb callback. I hope the comment above is adequate.

Next remove the typedef around the `enum uc_mem_type`. The typedef is now a forword declaration in line 247.

Next add the callback enum for tlb lookups.

Next create a enum for the suported tlb modes. The mode can changed with `UC_CTL_TLB_TYPE`. `uc_ctl_tlb_mode` is a helper function for that.

Reply:
Sorry I didn't express myself clearly. I mean could you leave more inline comments here, mostly about usage and caveats?